### ISSUE TITLE
Error raised when using alias of callable TypeVar

### ISSUE THREAD

**Issue by falan** [microsoft/pyright#1011]
**Describe the bug**
Type alias is not working properly when using callable typevar

**To Reproduce**

```python
from typing import TypeVar, Callable

T = TypeVar("T")

F = Callable[[T], str]

a: F
b: Callable[[T], str]
```

**Expected behavior**
No error.

**Screenshots or Code**
![bug](https://user-images.githubusercontent.com/4638204/92355079-1f01af00-f116-11ea-8900-ebe3357d9fab.JPG)

**VS Code extension or command-line**
VS Code extension since 1.1.65. No error report on 1.1.64.

**Comment by erictraut** [comment]
This is the correct behavior. 

I presume that you have "strict" type checking mode enabled because the `reportMissingTypeArgument` diagnostic rule is disabled for "basic" type checking mode.

The `reportMissingTypeArgument` diagnostic rule checks for cases where a generic type alias is used without providing the appropriate type arguments. In your case, you have defined a generic type alias `F` that takes one type argument, but you haven't provided that type argument when you provided the type annotation for variable `a`. You would need to change your code to something like `F[int]` or `F[str]`. If you are using `F` within the context of a generic class or generic function, you can also use generic type variables associated with that context. For example:

```
T1 = TypeVar("T1")
class MyClass(Generic[S]):
   a: F[S]

T2 = TypeVar("T2")
def my_func(p1: T2):
    a: F[T2]
```

Pyright should also generate a separate error for the annotation you have provided for variable `b` because TypeVar `T` has no meaning in this context. I just added support for generating this error, and it will be in the next release of Pyright.
